distributed/client.py

```diff
@@ -2579,55 +2586,60 @@ def _graph_to_futures(
             if values:
                 dsk = subs_multiple(dsk, values)
 
-            d = {k: unpack_remotedata(v, byte_keys=True) for k, v in dsk.items()}
-            extra_futures = set.union(*[v[1] for v in d.values()]) if d else set()
-            extra_keys = {tokey(future.key) for future in extra_futures}
-            dsk2 = str_graph({k: v[0] for k, v in d.items()}, extra_keys)
-            dsk3 = {k: v for k, v in dsk2.items() if k is not v}
```

Comment:
It looks like this line wasn't properly tested before and got dropped.  See #4145

Reply:
Not sure sure about this now.  Longer discussion in #4145